lib/beaker/hypervisor/docker.rb

```diff
@@ -65,10 +65,24 @@ def provision
         end
 
         @logger.debug("Creating container from image #{image_name}")
-        container = ::Docker::Container.create({
-          'Image' => image_name,
-          'Hostname' => host.name,
-        })
+        image_hash = {
+           'Image' => image_name,
+           'Hostname' => host.name,
+        }
+
+        if options[:dockeropts]
+          options[:dockeropts].each do |k,v|
+            image_hash[k] = v
+          end
+        end
+
+        if host[:dockeropts]
```

Comment:
Hi @fiddyspence,

This will be a little more maintainable if you eliminate the duplicate code block.  You can do that with these lines changed to something like this:

    if host[:dockeropts] || options[:dockeropts]
      dockeropts = host[:dockeropts] ? host[:dockeropts] : options[:dockeropts]
      dockeropts.each do |k,v|

Also, please add a unit test, so that we can verify that we don't break this new functionality accidentally in the future.

Let me know if you have any questions about this, and we look forward to getting this fix in!

Thanks,
Kevin